core/src/main/kotlin/org/evomaster/core/search/gene/UUIDGene.kt

```diff
@@ -79,12 +80,19 @@ class UUIDGene(
 
     fun getValueAsUUID(): UUID = UUID(mostSigBits.value, leastSigBits.value)
 
-    override fun copyValueFrom(other: Gene) {
+    override fun copyValueFrom(other: Gene): Boolean {
         if (other !is UUIDGene) {
             throw IllegalArgumentException("Invalid gene type ${other.javaClass}")
         }
-        this.mostSigBits.copyValueFrom(other.mostSigBits)
-        this.leastSigBits.copyValueFrom(other.leastSigBits)
+        val current = copy()
+        val ok = this.mostSigBits.copyValueFrom(other.mostSigBits) && this.leastSigBits.copyValueFrom(other.leastSigBits)
+
+        if (!ok || !isLocallyValid()){
+            Lazy.assert { copyValueFrom(current) }
```

Comment:
see previous comment about this. apply also for all other occurrences of this code.
however, as it seems there is a bit of duplicated code, should it be put in a common function?

Reply:
thanks! modified now